### ISSUE TITLE
Edit template button fails WCAG SC 2.5.3: Label in name

### ISSUE THREAD

**Issue by joedolson** [WordPress/gutenberg#47335]
### Description

In full site editing, the button with the visible text 'Edit' in templates has the accessible name "Open the editor" (provided by an aria-label.)

The Label in Name rule requires that user interface components containing text include that text as part of the accessible name of the component. While this meets that rule from a string partial perspective, it does not from a user perspective, because 'edit' and 'editor' are different parts of speech. 

It is also always preferable that the accessible name begins with the visible text, and that the part of speech matches the visible text. In this control, the visible active verb & first word is 'Edit', but the accessible verb and first word is 'Open'.

Recommendation: remove the aria-label attribute. 

### Step-by-step reproduction instructions

Open the site Editor. 
Inspect the 'Edit' button 
Note that the aria-label is 'Open the Editor'.

### Environment info

Gutenberg 15.0

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by annezazu** [comment]
Thanks for already getting this underway, Carolina, and for testing, Joe!
